lib/SILOptimizer/Mandatory/PerformanceDiagnostics.cpp

```diff
@@ -810,15 +810,18 @@ class PerformanceDiagnosticsPass : public SILModuleTransform {
       SmallVector<SILFunction *, 8> constructorsAndDestructors;
 
       for (SILFunction &function : *module) {
-        auto func = function.getLocation().getAsASTNode<AbstractFunctionDecl>();
-        if (func) {
-          if (isa<DestructorDecl>(func) || isa<ConstructorDecl>(func)) {
-            constructorsAndDestructors.push_back(&function);
-            continue;
-          }
-          if (getMethodDispatch(func) == MethodDispatch::Class) {
-            vtableMembers.push_back(&function);
-            continue;
+        if (function.hasLocation()) {
```

Comment:
Can you add a short comment saying why there are functions with no locations and why it's okay to not check those?